### ISSUE TITLE
Keep a local list of peers to facilitate bootstrapping upon reboot

### ISSUE THREAD

**Issue by jeremyid** [keep-network/keep-core#1856]
@mhluongo said on Discord that "It would be really nice to have some way to save peers between restarts".

Indeed it would, as this would both limit the risk of eclipse attacks as underlined by Matt, as well as prevent issues such as outdated bootstrap peers list in the config file that would require manual update.

One way to do it would be for the node to keep a routing table in the persistence folder and use it as a bootstrap list.

**Comment by jeremyid** [comment]
I have identified a possible way to implement this for this. I would like to do this first for the ECDSA client. I know which Go routines I want to change - however I need help setting up a local environment to build everything and test my changes as I am new to this and the documentation hasn't helped for what I want to do.

@mhluongo Would anyone from the team be willing to provide some help, perhaps through a call with screen sharing or something of the sort?

**Comment by mhluongo** [comment]
I don't think we could do 1:1 for a bit, considering deadlines for existing work. We're happy to help here or we could maybe do a group dev call to make the time higher leverage

**Comment by jeremyid** [comment]
I'm fine with a group dev call. Anything to make it clearer to community devs how to run things locally to test various kinds of changes. I think a few other people would be interested, maybe we could gauge interest on Discord.

**Comment by mhluongo** [comment]
cc @Shadowfiend @sthompson22 ideas or pointers on a better local setup?

@jeremyid anything you can share about your issues here would help!

**Comment by jeremyid** [comment]
I think I am almost there thanks to help from various community members. 

What I'm trying to do involves changes in both ``keep-core/.../libp2p.go`` and ``keep-ecdsa/.../start.go``. What I did:

- Forked both repos
- Added an instruction the ``replace``of ``keep-ecdsa``'s ``go.mod`` to point to my own forked repo: ``github.com/keep-network/keep-core => github.com/jeremyid/keep-core v.1.2.3-rc``
- Built the ECDSA docker image by running ``docker build .`` from ``keep-ecdsa`` 

I was able to run this image after creating the ``config.toml`` and other appropriate files / folders. 

The only thing left to do is figuring out how to work with ``v1.0.0`` instead of ``v1.1.1-rc`` which is the last release that was available when I forked my repo. I believe this is a noob Git problem though so I can probably figure it out on my own.

**Comment by mhluongo** [comment]
On the last point, branching off a tag would be the way to go. I'm not certain you should be off `v1.0.0` though- @sthompson22 what are the bootstraps running?

**Comment by jeremyid** [comment]
My own ECDSA Ropsten node as well as the bootstrap list I put in the ``config.toml`` are running ``v1.0.0`` afaik, althought it is my understanding that new contracts will be communicated soon for use with new clients. 

I think I managed to branch off the ``v1.0.0`` tag. Currently working on [this](https://github.com/jeremyid/keep-ecdsa/tree/last-peers-list) local branch with the change I mentioned to the ``replace`` statement in the ``go.mod``. I tried to compile with ``docker build .`` but got the following error:

![image](https://user-images.githubusercontent.com/14246657/84921091-50914e00-b0c4-11ea-9eb7-a4808c80923d.png)

Could this be a compatibility problem between the new ``keep-core v.1.2.3-rc`` and the old ``keep-ecdsa v1.0.0``?

**Comment by jeremyid** [comment]
I have solved this problem. Now proceeding with implementation.

**Comment by lionakhnazarov** [comment]
Closing as stale. Targets the legacy keep-core v1 (random beacon / KEEP staking) architecture, superseded by the Threshold v2 stack. Not actionable against current main.
